types/gulp-pug-linter/index.d.ts

```diff
@@ -0,0 +1,20 @@
+// Type definitions for gulp-tap 1.0
+// Project: https://github.com/ilyakam/gulp-pug-linter
+// Definitions by: Takesi Tokugawa <https://github.com/TokugawaTakesi>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+interface GulpPugLinter {
+    (options?: GulpPugLinter.Options): NodeJS.ReadWriteStream;
+}
+
+declare namespace GulpPugLinter {
+    interface Options {
+        reporter?: any;
```

Comment:
As far as I know, `any` is bad practice. In reality, `reporter` could only be a `function` or a `string`, as seen in [this section of the README](https://github.com/ilyakam/gulp-pug-linter#external-reporter).

Reply:
so, `string | (...args: any[]) => any` or maybe `string | Function` ?

Reply:
@ilyakam, linter will not pass `Function`  type. I suppose, it we don't know at advance the possible types of function arguments and type of returning value, `any` is inevitable.

Reply:
`(...args: any[]) => any` works for any function that Typescript has a type for. It doesn't work for variables with the explicit type `Function`, though.

I think this is good enough to merge, but you should test the above type and see if it works well enough, then submit a follow-up PR.